ui/jobs/event_emitter.ts

```diff
@@ -121,15 +121,11 @@ export class JobsEventEmitter extends EventEmitter<EventMap> {
   }
 
   processEnmityTargetData({ Target: target }: OverlayEventResponses['EnmityTargetData']): void {
-    if (target) {
-      this.emit('battle/target', {
-        name: target.Name,
-        distance: target.Distance,
-        effectiveDistance: target.EffectiveDistance,
-      });
-    } else {
-      this.emit('battle/target');
-    }
+    this.emit('battle/target', {
+      name: target.Name,
+      distance: target.Distance,
+      effectiveDistance: target.EffectiveDistance,
+    });
```

Comment:
If this is the case, then the underlying event definition in `event.d.ts` is wrong and should be corrected instead, which will then require the null check here as appropriate.

Said underlying event definition is also missing entries for `Focus` and `Hover`, if we're updating it we should probably update it to fully match the spec?

Reply:
Agree to update the definition, also I think the null check here is needed.

Reply:
See latest commit, event typedef is updated to match the C# definition, and null check is in place.
Distance is string and was just working because of type coercion before, so I elected to add `parseFloat`s to match the expected typing.

![image](https://user-images.githubusercontent.com/6119598/168315181-bbc17bdd-c93c-480b-b921-f37a265d759c.png)